addons/hr_expense/models/hr_expense.py

```diff
@@ -907,6 +918,13 @@ def _compute_attachment_number(self):
         for sheet in self:
             sheet.attachment_number = sum(sheet.expense_line_ids.mapped('attachment_number'))
 
+    @api.depends('company_id.currency_id')
+    def _compute_currency(self):
+        for sheet in self:
+            # Deal with a display bug when there is a company currency change after creation of the expense sheet
+            if not sheet.currency_id or sheet.state not in {'post', 'done', 'cancel'}:
+                sheet.currency_id = sheet.company_id.currency_id
```

Comment:
I have the feeling you could change the currency on-the-fly and make something that doesn't have any sense with that

Reply:
That just make sure that if the sheet is not linked to any move yet and you change the company currency, the sheet currency (which is related to the company currency but is stored so if you change the company currency it doesn't update properly the currency, do you have a better idea on how to deal with that?)
It's a niche case, mostly impacting demo data

Reply:
As I said, it's mostly about the demo data going crazy due to the main company changing currency after the expenses are created (without any moves posted)
You're right, it's impossible to change the company currency as soon as some accounting data has been created